src/plugins/data_view_editor/public/components/preview_panel/preview_panel.tsx

```diff
@@ -86,7 +90,7 @@ export const PreviewPanel = ({ type, allowHidden, title = '', matchedIndices$ }:
         query={title}
       />
       <EuiSpacer size="m" />
-      {Boolean(title) && currentlyVisibleIndices.length > 0 && (
+      {Boolean(title) && (
```

Comment:
Thanks again for the contribution @lsq645599166, and apologies for all the back and forth on this. After testing this locally, I think misunderstood the intent of the code originally. I also overlooked @kertal's second comment where he suggested hiding the tabs entirely, but I agree this is actually the best approach.

We already do this before the user has typed anything, and we show the full list of sources:
![image](https://github.com/user-attachments/assets/5796d5d4-7fb0-43a9-b220-4390abb2132a)

We also intentionally return the full list as `visibleIndices` from `getMatchedIndices` when there are no matches because we want the user to see the available options when their pattern doesn't match. It turns out this was behaving as intended, but the confusing part was that the tabs were still visible and "Matching sources" was still selected.

If we hide the tabs when there are no matches as @kertal suggested, it behaves the same as if the user hadn't typed anything, so there's no confusion about the tabs, and they'll still be able to see the available sources:
![image](https://github.com/user-attachments/assets/439e6187-dcb7-4da4-94b7-69116e17e691)

This seems like the least confusing and most consistent behaviour (and matches the original intent). Luckily if we follow the same logic used in `src/plugins/data_view_editor/public/components/preview_panel/status_message/status_message.tsx`, I think we should be able to revert all changes except for this line in `PreviewPanel`.
```suggestion
      {title.length > 0 && (matched.exactMatchedIndices.length > 0 || matched.partialMatchedIndices.length > 0) && (
```

Reply:
Ok, I commit a change to hide tab when no matched index.
![image](https://github.com/user-attachments/assets/85e0b903-e5f9-4e1b-b414-e905d0189fb8)